contracts/v3/alchemix/Alchemist.sol

```diff
@@ -0,0 +1,847 @@
+    ///
+    /// An approval is required to transfer native tokens to the transmuter.
+    function repay(uint256 _parentAmount, uint256 _childAmount)
+        external
+        nonReentrant
+        noContractAllowed
+        onPriceCheck
+        expectInitialized
+    {
+        CDP.Data storage _cdp = _cdps[msg.sender];
+        _cdp.update(_ctx);
+
+        if (_parentAmount > 0) {
+            token.safeTransferFrom(msg.sender, address(this), _parentAmount);
+            _distributeToTransmuter(_parentAmount);
+        }
+
+        if (_childAmount > 0) {
+            xtoken.burnFrom(msg.sender, _childAmount);
+            //lower debt cause burn
+            xtoken.lowerHasMinted(_childAmount);
+        }
+
+        uint256 _totalAmount = _parentAmount.add(_childAmount);
+        _cdp.totalDebt = _cdp.totalDebt.sub(_totalAmount, '');
+
+        emit TokensRepaid(msg.sender, _parentAmount, _childAmount);
+    }
+
+    /// @dev Attempts to liquidate part of a CDP's collateral to pay back its debt.
+    ///
+    /// @param _amount the amount of collateral to attempt to liquidate.
+    function liquidate(uint256 _amount)
+        external
+        nonReentrant
+        noContractAllowed
+        onPriceCheck
+        expectInitialized
+        returns (uint256, uint256)
+    {
+        CDP.Data storage _cdp = _cdps[msg.sender];
+        _cdp.update(_ctx);
+
+        // don't attempt to liquidate more than is possible
+        if (_amount > _cdp.totalDebt) {
+            _amount = _cdp.totalDebt;
+        }
+        (uint256 _withdrawnAmount, uint256 _decreasedValue) = _withdrawFundsTo(
+            address(this),
+            _amount
+        );
+        //changed to new transmuter compatibillity
+        _distributeToTransmuter(_withdrawnAmount);
+
+        _cdp.totalDeposited = _cdp.totalDeposited.sub(_decreasedValue, '');
+        _cdp.totalDebt = _cdp.totalDebt.sub(_withdrawnAmount, '');
+        emit TokensLiquidated(msg.sender, _amount, _withdrawnAmount, _decreasedValue);
+
+        return (_withdrawnAmount, _decreasedValue);
+    }
+
+    /// @dev Mints synthetic tokens by either claiming credit or increasing the debt.
+    ///
+    /// Claiming credit will take priority over increasing the debt.
+    ///
+    /// This function reverts if the debt is increased and the CDP health check fails.
+    ///
+    /// @param _amount the amount of alchemic tokens to borrow.
+    function mint(uint256 _amount)
+        external
+        nonReentrant
+        noContractAllowed
+        onPriceCheck
+        expectInitialized
+    {
+        CDP.Data storage _cdp = _cdps[msg.sender];
+        _cdp.update(_ctx);
+
+        uint256 _totalCredit = _cdp.totalCredit;
+
+        if (_totalCredit < _amount) {
+            uint256 _remainingAmount = _amount.sub(_totalCredit);
+            _cdp.totalDebt = _cdp.totalDebt.add(_remainingAmount).add(
+                _remainingAmount.mul(borrowFee).div(PERCENT_RESOLUTION)
```

Comment:
The Debt is increased with _remainingAmount.mul(borrowFee).div(PERCENT_RESOLUTION)
However nothing else is done with this amount.
For comparison function harvest() has this call to do something with the fees:
```JS
if (_feeAmount > 0) {
      token.safeTransfer(rewards, _feeAmount);
}

```

Reply:
The debt state is just book keeping

Reply:
@gpersoon I have updated the code. The fee will be sent to reward address now

Reply:
I think the number of xtoken send should also be subtracted with the _borrowFeeAmount
something like:
_amount = _amount.sub(__borrowFeeAmount);
Otherwise, who is paying for the fees?
(please doublecheck if my hunch is correct)

Reply:
_amount should be the same. 

> `Otherwise, who is paying for the fees?`

The fee is paid by the user. The user needs to clear debt to get its collateral back